### PR TITLE
Refactor some misuse of RwLock fields on the butterfly Server struct

### PR DESCRIPTION
**Pull request by baumanj**
- Remove unnecessary `Arc<RwLock<_>`> around `SocketAddr` fields that were never mutated
- Change read-only `block_list` access to use `RwLock::read`, not `::write`
- Return `SocketAddr` references rather than owned values to avoid cloning
- Make `swim_port()` and `gossip_port()` private
- Change `dat_file` RwLock to Mutex since there is no read/write distinction and move the `Option` to the outside so we don't incur the unlock cost to check if the option is some

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/butterfly/src/server/mod.rs [hunk 1]
```diff
@@ -321,11 +321,11 @@ impl Server {
...
                     update_store: RumorStore::default(),
                     departure_store: RumorStore::default(),
-                    swim_addr: Arc::new(RwLock::new(swim_socket_addr)),
-                    gossip_addr: Arc::new(RwLock::new(gossip_socket_addr)),
+                    swim_addr: swim_socket_addr,
+                    gossip_addr: gossip_socket_addr,
                     suitability_lookup: Arc::new(suitability_lookup),
                     data_path: Arc::new(data_path.as_ref().map(|p| p.into())),
-                    dat_file: Arc::new(RwLock::new(None)),
+                    dat_file: None,
```
Comment (christophermaier) [inline components/butterfly/src/server/mod.rs]: tenor-173784934

**Pull request merged**